src/components/FileDownloaderLink.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useRef, useEffect } from 'react'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faFileCsv } from '@fortawesome/free-solid-svg-icons'
+
+interface FileSaverLinkProps {
+  data?: string
+  options?: BlobPropertyBag
+}
+
+export const FileDownloaderLink: React.FC<FileSaverLinkProps> = ({ data, options }) => {
+  const lastObjectURL = useRef('')
+  const lastDataUsed = useRef('')
+
+  useEffect(
+    () => (): void => {
+      // on unmount release ObjectURL if any
+      if (lastObjectURL.current) URL.revokeObjectURL(lastObjectURL.current)
+    },
+    [],
+  )
+
+  // only create download onClick because
+  // creating and retaining BLob and ObjectURL can be expensive on each render
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
+    // no data
+    if (!data) {
+      // prevent invalid file download
+      e.preventDefault()
+      return
+    }
+    // data didn't change, reusing already set ObjectURL
+    if (lastDataUsed.current === data) return
+    lastDataUsed.current = data
+
+    // new data, need new ObjectURL
+    // release old if was set previously
+    if (lastObjectURL.current) URL.revokeObjectURL(lastObjectURL.current)
+
+    const blob = new Blob([data], options)
+    e.currentTarget.href = lastObjectURL.current = URL.createObjectURL(blob)
+  }
+
+  return (
+    <a href={lastObjectURL.current} download="trades.csv" onClick={handleClick}>
```

Comment:
Idea, not for now necessarily.

Take the file name as an input parameter. 
We might want to use this for downloading orders as well.

Also, thinking about adding address and date to file name...